### ISSUE TITLE
Inference - different output when using different batch size

### ISSUE THREAD

**Issue by thariq-nugrohotomo** [facebookresearch/LeViT#28]
When performing inference using pretrained model (within `eval()` mode), the same image may producing different logits output when batch size is changed. 

Example code: 
```python
with torch.no_grad():
    x = torch.stack([
        torch.zeros((3,224,224)),
        torch.ones((3,224,224)),
        torch.ones((3,224,224)),
    ])
    model = LeViT_384(pretrained=True)
    model = model.eval()
    print('batch=1', model(x[:1])[0][:2].numpy())
    print('batch=3', model(x[:3])[0][:2].numpy())
```

Output: (only for the 1st sample, limited to 2 classes)
```
batch=1 [-0.3287484  -0.11664876]
batch=3 [-0.32874817 -0.11664899]
```

While the argmax may not significantly affected, this inconsistency make it difficult to perform gradient analysis.

I'm suspecting that this is caused by some batch-normalization layers not honoring `eval()` mode.

**Comment by btgraham** [comment]
To calculate gradients with batchnorm turned off, you can clone the model and use utils.replace_batchnorm to put the model into 'eval' mode. (The code is not really designed to do this, as I never needed gradients in eval mode).
